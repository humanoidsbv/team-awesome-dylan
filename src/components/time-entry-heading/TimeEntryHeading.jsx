import React from 'react';
import PropTypes from 'prop-types';

import styles from './TimeEntryHeading.module.css';

const TimeEntryHeading = ({ startDate }) => (
  <div className={styles.timeStamp}>
    <span>{startDate}</span>
    <span className={styles.duration}>08:00:00 </span>
  </div>
);

TimeEntryHeading.propTypes = {
  startDate: PropTypes.string.isRequired
};

export default TimeEntryHeading;
