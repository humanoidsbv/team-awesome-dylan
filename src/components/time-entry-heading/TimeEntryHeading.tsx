import React from 'react';

import styles from './TimeEntryHeading.module.css';
import { TimeEntryInterface } from '../../redux/time-entries/types';

const TimeEntryHeading = ({ startTimestamp }: TimeEntryInterface): React.ReactElement => {
  const headerString = new Date(startTimestamp).toLocaleDateString('nl-NL', {
    day: 'numeric',
    month: 'long',
    weekday: 'long'
  });

  return (
    <div className={styles.timeStamp}>
      <span>{headerString}</span>
      <span className={styles.duration}>08:00:00 </span>
    </div>
  );
};

export default TimeEntryHeading;
