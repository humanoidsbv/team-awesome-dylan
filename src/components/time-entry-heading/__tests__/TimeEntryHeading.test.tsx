import * as React from 'react';
import { shallow } from 'enzyme';

import TimeEntryHeading from '../TimeEntryHeading';

it('matches snapshot', () => {
  const timeEntryHeading = shallow(
    <TimeEntryHeading
      startTimestamp='2019-10-16T10:00:00.000Z'
    />
  );
  expect(timeEntryHeading).toMatchSnapshot();
});
