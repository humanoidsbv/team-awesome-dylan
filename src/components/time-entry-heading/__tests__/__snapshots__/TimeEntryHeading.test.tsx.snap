// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="timeStamp"
>
  <span>
    woensdag 16 oktober
  </span>
  <span
    className="duration"
  >
    08:00:00 
  </span>
</div>
`;
